### ISSUE TITLE
freelan.cfg is overwhelming

### ISSUE THREAD

**Issue by leggewie** [freelan-developers/freelan#151]
freelan.cfg is overwhelmingly complex.  I think it should only host the most important configurations to get freelan running quickly. Anything to be considered "tinkering" should be configured in another file.  Any optional configuration should be in another file or moved to the end of freelan.cfg and clearly labeled optional.

The user should find at the top two or three knobs he needs to configure and then be done with it.  Otherwise, I assume the majority of people testing freelan out will throw it out the window as too complex.  Freelan at the moment is too technical and too complex IMHO both in its docs and communication of what it is to potential users.  It is highly configurable and that is great.  But we should not bother the ordinary user with it.  Set sane defaults and only require the user to configure the absolute bare minimum.

**Comment by s-vincent** [comment]
Yes I agree, configuration file could be very discouraging for beginner but with good documentation like a quick start guide it could be easily handled.

Here are the minimum options user has to set to use FreeLAN with X509-based authentication:
 * fscp.contact;
 * tap_adapter.ipv4_address_prefix_length.
 * tap_adapter.ipv6_address_prefix_length.
 * security.signature_certificate_file;
 * security.signature_private_key_file;
 * security.authority_certificate_file.

Here are the minimum options user has to set to use FreeLAN with PSK-based authentication:
 * fscp.contact;
 * tap_adapter.ipv4_address_prefix_length;
 * tap_adapter.ipv6_address_prefix_length;
 * security.passphrase;
 * security.passphrase_salt.

**Comment by leggewie** [comment]
For the config file that we expect anyone to touch, I think we should focus only on

- PSK
- IPv4
- no salt

IMO, the user should supply a PW and a remote host/IP, nothing else.  As long as ports are opened/can be opened by UPnP, the goal should be for the user to be up and running after no more than 2 minutes.  Alterations and gradual tweaks can come after this initial 'success'.

Anybody who wants certs and stuff will be willing to read the necessary docs and spend the necessary time.  Likely, they will be higher skilled, too.

**Comment by s-vincent** [comment]
@leggewie What do you think about having two "simple" example files (for Alice and Bob) in repository with three options you give and without any others or comments?

Users will pick the file as is, see just the very simple option and try.

**Comment by richman1000000** [comment]
I agree that freelan.cfg should be split-able.
for example place into file **blocked.cfg**
> [fscp]
> never_contact=haker.ru

and then write into freelan.cfg

> include blocked.cfg

and those options are applied

**Comment by richman1000000** [comment]
from HelpDesk worked point of  view  freelan.cfg is really complicated.
simple freelan config looks this 
> include freelan.advance.cfg
> include  freelan.user.cfg.

Worker willopen  freelan.user.cfg and see only.
> tap_adapter.ipv4_address_prefix_length.
> security.signature_certificate_file;
> security.signature_private_key_file;

**Comment by richman1000000** [comment]
notice that I did not included **tap_adapter.ipv6_address_prefix_length** in simple example cause if worker is needs it - it will be included.

**Comment by leggewie** [comment]
@s-vincent Can you provide an example?
